Makefile fails on macOS ARM (M1/M2/M3) when running 'make kind-create'

**Describe the bug**

When running the command make kind-create for the first time on MacBook models with Apple Silicon processors (M1, M2, or M3), the process fails with the following error:

```
chmod: /Users/.../kind-v0.20.0: No such file or directory
make: *** [kind-create] Error 2
```
The Makefile tries to download and execute the kind binary, but it fails because it only checks for aarch64 (used on Linux ARM systems) and does not recognize macOS ARM (arm64).
As a result, the binary is never downloaded, and chmod fails due to a missing file.

**To Reproduce**

Steps to reproduce the behavior:

1. Use one of the following models:

- MacBook Air (M1, 2020)
- MacBook Pro (M1/M2/M3, 2021–2024)
- Mac mini (M1/M2)
- Mac Studio (M2/M3)

2. Clone the repository. 
3. Run the command:

`make kind-create`


4. Observe the error:

```
chmod: /Users/.../kind-v0.20.0: No such file or directory
make: *** [kind-create] Error 2

```
**Expected behavior**

The Makefile should correctly detect macOS (Darwin) with ARM64 architecture
and download the appropriate binary kind-darwin-arm64 instead of the Linux binary kind-linux-arm64.

**Screenshots**

<img width="1134" height="738" alt="Image" src="https://github.com/user-attachments/assets/35b5fdaa-4e02-436e-82c7-b6c7fc414988" />

--- a/make/helm.mk
+++ b/make/helm.mk
@@ -4,8 +4,40 @@ HELM_VERSION = v3.15.0
 HELM_V_BINARY := $(LOCALBIN)/helm-$(HELM_VERSION)
 
 $(HELM_V_BINARY): $(LOCALBIN)  ## Installs helm in $PROJECT_DIR/bin
-# For AMD64 / x86_64
-ifeq ($(shell uname -p),x86_64)
+# Download helm binary depending on architecture and OS
+ifeq ($(shell uname -s),Darwin)
+  # macOS
+  ifeq ($(shell uname -m),arm64)
+	@{ \
+	set -e ;\
+	curl -Lo helm.tar.gz https://get.helm.sh/helm-$(HELM_VERSION)-darwin-arm64.tar.gz ;\
+	tar -zxvf helm.tar.gz ;\
+	mv darwin-arm64/helm $@ ;\
+	chmod +x $@ ;\
+	rm -rf darwin-arm64 helm.tar.gz ;\
+	}
+  else
+	@{ \
+	set -e ;\
+	curl -Lo helm.tar.gz https://get.helm.sh/helm-$(HELM_VERSION)-darwin-amd64.tar.gz ;\
+	tar -zxvf helm.tar.gz ;\
+	mv darwin-amd64/helm $@ ;\
+	chmod +x $@ ;\
+	rm -rf darwin-amd64 helm.tar.gz ;\
+	}
+  endif
+else
+  # Linux
+  ifeq ($(shell uname -m),aarch64)
+	@{ \
+	set -e ;\
+	curl -Lo helm.tar.gz https://get.helm.sh/helm-$(HELM_VERSION)-linux-arm64.tar.gz ;\
+	tar -zxvf helm.tar.gz ;\
+	mv linux-arm64/helm $@ ;\
+	chmod +x $@ ;\
+	rm -rf linux-arm64 helm.tar.gz ;\
+	}
+  else
 	@{ \
 	set -e ;\
 	curl -Lo helm.tar.gz https://get.helm.sh/helm-$(HELM_VERSION)-linux-amd64.tar.gz ;\
@@ -14,18 +46,9 @@ ifeq ($(shell uname -p),x86_64)
 	chmod +x $@ ;\
 	rm -rf linux-amd64 helm.tar.gz ;\
 	}
+  endif
 endif
-# For ARM64
-ifeq ($(shell uname -p),aarch64)
-	@{ \
-	set -e ;\
-	curl -Lo helm.tar.gz https://get.helm.sh/helm-$(HELM_VERSION)-linux-arm64.tar.gz ;\
-	tar -zxvf helm.tar.gz ;\
-	mv linux-arm64/helm $@ ;\
-	chmod +x $@ ;\
-	rm -rf linux-amd64 helm.tar.gz ;\
-	}
-endif
+
 
 .PHONY: helm
 helm: $(HELM_V_BINARY) ## Download helm locally if necessary.

--- a/make/kind.mk
+++ b/make/kind.mk
@@ -9,13 +9,21 @@ KIND_V_BINARY := $(LOCALBIN)/kind-$(KIND_VERSION)
 kind: $(KIND_V_BINARY)
 
 $(KIND_V_BINARY): $(LOCALBIN)  ## Installs kind in $PROJECT_DIR/bin
-# For AMD64 / x86_64
-ifeq ($(shell uname -p),x86_64)
-	curl -Lo $@ https://kind.sigs.k8s.io/dl/$(KIND_VERSION)/kind-linux-amd64
-endif
-# For ARM64
-ifeq ($(shell uname -p),aarch64)
+# Download kind binary depending on architecture and OS
+ifeq ($(shell uname -s),Darwin)
+  # macOS
+  ifeq ($(shell uname -m),arm64)
+	curl -Lo $@ https://kind.sigs.k8s.io/dl/$(KIND_VERSION)/kind-darwin-arm64
+  else
+	curl -Lo $@ https://kind.sigs.k8s.io/dl/$(KIND_VERSION)/kind-darwin-amd64
+  endif
+else
+  # Linux
+  ifeq ($(shell uname -m),aarch64)
 	curl -Lo $@ https://kind.sigs.k8s.io/dl/$(KIND_VERSION)/kind-linux-arm64
+  else
+	curl -Lo $@ https://kind.sigs.k8s.io/dl/$(KIND_VERSION)/kind-linux-amd64
+  endif
 endif
 	chmod +x $@
 

